xfer/publisher.go

```diff
@@ -62,6 +62,27 @@ func (p *TCPPublisher) loop(incoming <-chan net.Conn) {
 				return // someone closed our connection chan -- weird?
 			}
 
+			// Don't allow multiple connections from the same remote host.
+			host, _, err := net.SplitHostPort(conn.RemoteAddr().String())
+			if err != nil {
+				log.Printf("incoming connection: %s: %v (dropped)", conn.RemoteAddr(), err)
+				conn.Close()
+				continue
+			}
+		outer:
+			for activeConn := range activeConns {
+				activeHost, _, err := net.SplitHostPort(activeConn.RemoteAddr().String())
+				if err != nil {
+					log.Printf("active connection: %s: %v (strange)", activeConn.RemoteAddr(), err)
+					continue
+				}
+				if host == activeHost {
+					log.Printf("duplicate connection: %s (dropped)", conn.RemoteAddr())
+					conn.Close()
+					continue outer
+				}
+			}
+
```

Comment:
This is a bit ugly; perhaps a new struct connection containing the encoder and the conn, and then key the above active map with the host from above?